File: sample/cs/04_monte_carlo_pi.cs
using System.Collections.Generic;
using System.IO;
using System;

public static class Program
{
    public static long lcg_next(long state)
    {
        return (((1664525L * state) + 1013904223L) % 4294967296L);
    }

    public static double run_pi_trial(long total_samples, long seed)
    {
        long inside = 0L;
        long state = seed;
        var __pytra_range_start_1 = 0;
        var __pytra_range_stop_2 = total_samples;
        var __pytra_range_step_3 = 1;
        if (__pytra_range_step_3 == 0) throw new Exception("range() arg 3 must not be zero");
        for (var __pytra_unused_4 = __pytra_range_start_1; (__pytra_range_step_3 > 0) ? (__pytra_unused_4 < __pytra_range_stop_2) : (__pytra_unused_4 > __pytra_range_stop_2); __pytra_unused_4 += __pytra_range_step_3)
        {
            state = lcg_next(state);
            double x = ((double)(state) / (double)(4294967296.0));
            state = lcg_next(state);
            double y = ((double)(state) / (double)(4294967296.0));
            double dx = (x - 0.5);
            double dy = (y - 0.5);
            if (Pytra.CsModule.py_runtime.py_bool((((dx * dx) + (dy * dy)) <= 0.25)))
            {
                inside = (inside + 1L);
            }
        }
        return ((double)((4.0 * inside)) / (double)(total_samples));
    }

    public static void run_monte_carlo_pi()
    {
        long samples = 54000000L;
        long seed = 123456789L;
        double start = Pytra.CsModule.time.perf_counter();
        double pi_est = run_pi_trial(samples, seed);
        double elapsed = (Pytra.CsModule.time.perf_counter() - start);
        Pytra.CsModule.py_runtime.print("samples:", samples);
        Pytra.CsModule.py_runtime.print("pi_estimate:", pi_est);
        Pytra.CsModule.py_runtime.print("elapsed_sec:", elapsed);
    }

    public static void Main(string[] args)
    {
        run_monte_carlo_pi();
    }
}

File: sample/rs/04_monte_carlo_pi.rs
#[path = "../../src/rs_module/py_runtime.rs"]
mod py_runtime;
use py_runtime::{math_cos, math_exp, math_floor, math_sin, math_sqrt, perf_counter, py_bool, py_grayscale_palette, py_in, py_isalpha, py_isdigit, py_len, py_print, py_save_gif, py_slice, py_write_rgb_png};

// このファイルは自動生成です（native Rust mode）。

fn lcg_next(mut state: i64) -> i64 {
    return ((((((1664525) * (state))) + (1013904223))) % (4294967296));
}

fn run_pi_trial(mut total_samples: i64, mut seed: i64) -> f64 {
    let mut inside: i64 = 0;
    let mut state: i64 = seed;
    for _ in (0)..(total_samples) {
        state = lcg_next(state);
        let mut x: f64 = ((( state ) as f64) / (( 4294967296.0 ) as f64));
        state = lcg_next(state);
        let mut y: f64 = ((( state ) as f64) / (( 4294967296.0 ) as f64));
        let mut dx: f64 = ((x) - (0.5));
        let mut dy: f64 = ((y) - (0.5));
        if py_bool(&(((((((dx) * (dx))) + (((dy) * (dy))))) <= (0.25)))) {
            inside = inside + 1;
        }
    }
    return ((( (((( 4.0 ) as f64) * (( inside ) as f64))) ) as f64) / (( total_samples ) as f64));
}

fn run_monte_carlo_pi() -> () {
    let mut samples: i64 = 54000000;
    let mut seed: i64 = 123456789;
    let mut start: f64 = perf_counter();
    let mut pi_est: f64 = run_pi_trial(samples, seed);
    let mut elapsed: f64 = ((perf_counter()) - (start));
    println!("{} {}", "samples:".to_string(), samples);
    println!("{} {}", "pi_estimate:".to_string(), pi_est);
    println!("{} {}", "elapsed_sec:".to_string(), elapsed);
}

fn main() {
    run_monte_carlo_pi();
}
